tensorflow/core/common_runtime/gpu/gpu_util.cc

```diff
@@ -230,7 +235,9 @@ void GPUUtil::DeviceToDeviceCopy(
     // truly free.
     // TODO(zhengxq): remove this dependency when we switch to a better way
     // to make sure the memory is free.
-    send_device_to_device_stream->ThenWaitFor(recv_stream);
+    if (send_device_to_device_stream != recv_stream) {
```

Comment:
```suggestion
    if (!src->merge_device_to_device_stream()) {
```

Reply:
We cannot use `if (!src->merge_device_to_device_stream())`  here because when `src->merge_device_to_device_stream() == true`, it's still necessary to execute `send_device_to_device_stream->ThenWaitFor(recv_stream)` if the sending and receiving are from two GPU devices. So using `if (send_device_to_device_stream != recv_stream)` is the most accurate way. We clarified this in the comment for `StreamMergeOptions::merge_device_to_device_stream` in `config.proto`.